src/lib/journey/_utilities/metadata.utilities.test.ts

```diff
@@ -171,10 +171,13 @@ describe('Test metadata builder function for step', () => {
       ],
     };
     const result = buildStepMetadata(callbackMetadata, stageJson);
-
-    expect(result).toStrictEqual({
+    expect(result).toEqual({
       derived: {
-        isStepSelfSubmittable: false,
+        // because this is a a function now, i'm using the function defined so
+        // the memory is the same and its picked up as such by toEqual
+        // rather than creating a new function which would be different
+        // in memory.
```

Comment:
If this comment is going to persist in our source code, let's ensure it's written well.

```
/**
 * Unlike the other properties, `isStepSelfSubmittable` is a function,
 * so for `toEqual` to pass, the "expected" object needs to reference
 * the same function as the `result` object.
 *
 * TODO: Modify this to assert against the returned value, rather than
 * relying on function reference.
 */
```